test/static_arrays.jl

```diff
@@ -17,7 +17,7 @@ for alg in (nothing, LUFactorization(), SVDFactorization(), CholeskyFactorizatio
     sol = solve(LinearProblem(A, b), alg)
     @inferred solve(LinearProblem(A, b), alg)
     @test norm(A * sol .- b) < 1e-10
-    if alg <: KrylovJL_GMRES
```

Comment:
```suggestion 
    if alg isa KrylovJL_GMRES
```

Reply:
That's what I had first https://github.com/SciML/LinearSolve.jl/pull/518/commits/26574e6e155bf7da8033a60338c7f13ac6546787

Reply:
If `a isa T` is like `typeof(a) <: T`, but the former is preferred.
If `a` isn't a type, `a <: T` doesn't make sense and will throw an error.
https://github.com/SciML/LinearSolve.jl/actions/runs/10153821474/job/28077889265?pr=518#step:6:1114